modules/crawlerTest/src/org/labkey/crawlertest/CrawlerTestController.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2020 LabKey Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.labkey.crawlertest;
+
+import org.labkey.api.action.LabKeyError;
+import org.labkey.api.action.SimpleErrorView;
+import org.labkey.api.action.SimpleViewAction;
+import org.labkey.api.action.SpringActionController;
+import org.labkey.api.module.ModuleLoader;
+import org.labkey.api.security.RequiresPermission;
+import org.labkey.api.security.permissions.ReadPermission;
+import org.labkey.api.util.ErrorView;
+import org.labkey.api.view.JspView;
+import org.labkey.api.view.NavTree;
+import org.springframework.validation.BindException;
+import org.springframework.web.servlet.ModelAndView;
+
+public class CrawlerTestController extends SpringActionController
+{
+    private static final DefaultActionResolver _actionResolver = new DefaultActionResolver(CrawlerTestController.class);
+    public static final String NAME = "crawlertest";
+
+    public CrawlerTestController()
+    {
+        setActionResolver(_actionResolver);
+    }
+
+    @RequiresPermission(ReadPermission.class)
+    public class InjectJspAction extends SimpleViewAction<InjectForm>
+    {
+        @Override
+        public ModelAndView getView(InjectForm form, BindException errors)
+        {
+            if (!getViewContext().getContainer().getActiveModules().contains(ModuleLoader.getInstance().getModule(CrawlerTestModule.class)))
```

Comment:
I like trying to be conservative here. Perhaps even safer and easier for the test code would be to have a short allowable list of script injection parameter values that the test uses. Everything else could get ignored, and we wouldn't need to worry about the module's enabled status and would have zero worries about it causing problems on a production server.

Reply:
I updated the action to require dev mode and to only respond to a specific test user. I'd like to be able to experiment with other injection strings without having to rebuild the module.

Reply:
OK, sounds good. Restricting by user should make it impossible to "exploit" even if this module somehow got deployed to a production server.